cs/test/comm/Epoxy/EpoxyListenerTests.cs

```diff
@@ -114,13 +116,32 @@ public class EpoxyListenerTests : EpoxyTestBase
             var connection = await transport.ConnectToAsync(localhostAddress);
             await connection.StopAsync();
 
-            bool wasSignaled = disconnectedEventDone.Wait(TimeSpan.FromSeconds(30));
+            bool wasSignaled = disconnectedEventDone.Wait(TimeSpan.FromSeconds(10));
             Assert.IsTrue(wasSignaled, "Timed out waiting for Disconnected event to complete");
 
             Assert.IsNotNull(disconnectedConnection);
             Assert.AreEqual(connection.LocalEndPoint, disconnectedConnection.RemoteEndPoint);
         }
 
+        [Test]
+        public async Task OneConnectionStalledDuringHandshake_CanAcceptAnother()
+        {
+            EpoxyTransport transport = MakeTransport();
+            listener = transport.MakeListener(localhostEndpoint);
+            await listener.StartAsync();
+
+            var noHandshakeConnection = new TcpClient();
+            // This will just establish a TCP connection. It won't perform the Epoxy handshake, so
+            // the listener will just be sitting there waiting for the client to send some data.
+            await noHandshakeConnection.ConnectAsync(localhostEndpoint.Address, localhostEndpoint.Port);
+
+            var connectTask = transport.ConnectToAsync(localhostAddress);
+            bool didConnect = connectTask.Wait(TimeSpan.FromSeconds(10));
```

Comment:
I get the concern about load, but this still seems three orders of magnitude too big. If it's taking 100 ms to accept a connection, either we have a huge regression, or we might want to check out other CI services. Either way, I want to know about it. Also, this is going to be really annoying if this test is ever broken and you're debugging it - no test should block that long.

I won't speculate on the relative weights of these operations, but for what it's worth, I needed a delay to make metrics tests reliable. It's [50 ms](https://github.com/Microsoft/bond/blob/master/cs/test/comm/Interfaces/MetricsTests.cs#L23).